azure/functions/decorators/blob.py

```diff
@@ -17,7 +17,10 @@ def __init__(self,
                  **kwargs):
         self.path = path
         self.connection = connection
-        self.source = source.value if source else None
+        if type(source) is BlobSource:
+            self.source = source.value if source else None
+        else:
+            self.source = source

```

Comment:
If its a string, do we check for the case?

Reply:
Discussed, not needed. Will be recommended for users to use the enum instead. If they use the string version, the case will be recommended to be "EventGrid"